Makefile.am

```diff
@@ -11,18 +11,28 @@ dist_systemdunit_DATA = \
 	eos-live-boot-overlayfs-setup.service \
 	$(NULL)
 
-# Unfortunately, dist chokes on the escaped \x2d, so we need to
-# split this out into EXTRA_DIST unescaped.
+# Unfortunately, the escaped \x2d is handled poorly, so we distribute and
+# install this unit under another name and rename it by hand.
 systemdunit_DATA = \
-	var-endless\\x2dextra.mount
+	var-endless_x2dextra.mount
+
+install-data-hook:
+	pushd "$(DESTDIR)$(systemdunitdir)"; \
+	mv var-endless_x2dextra.mount var-endless\\x2dextra.mount; \
```

Comment:
I would quote the `\` escaped one so that you at least avoid the shell stripping one of the `\`s.